### ISSUE TITLE
getting different results tho nothing is changed

### ISSUE THREAD

**Issue by kmdrGroch** [istanbuljs/nyc#1154]
## Expected Behavior
Getting the same output if i dont change code

## Observed Behavior
Getting different results

## Environment Information
```
  System:
    OS: Windows 10
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
    Memory: 2.71 GB / 7.90 GB
  Binaries:
    Node: 10.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    nyc: ^14.0.0 => 14.1.1 
    source-map-support: ^0.5.12 => 0.5.12 
    ts-node: ^8.3.0 => 8.3.0 
    typescript: ^3.5.3 => 3.5.3

```
## Configuration
package.json
```
"nyc": {
    "extends": "@istanbuljs/nyc-config-typescript",
    "all": true,
    "check-coverage": true,
    "reporter": ["text", "html"]
  }
```
mocha.opts
```
--require ts-node/register/transpile-only
--require source-map-support/register
--recursive
test/**.ts
```
## Function I am testing
```typescript
type statusNames = "passes" | "both" | "fails";
interface Item {
  codes?: {
    [key in statusNames]?: string;
  };
  dependencies?: string[];
  description?: string;
  name: string;
}
interface Config {
  docker: boolean;
  items: Item[];
  language: string;
  projectName: string;
}
const phpParser = (code: Config) => {
  let newCode = `<?xml version="1.0" encoding="UTF-8"?>
<project name="${ code.projectName }" default="${ (code.items[0] || {}).name || "" }">`;

  if (code.docker) {
    newCode += `\n  <!-- Docker properties -->
  <property name="docker" value="docker" />
  <property name="dockerFilePath" value="\${basedir}/Dockerfile" />
  <property name="dockerBaseDir" value="\${basedir}" />`;
  }

  if (code.items.length > 0) {
    newCode += "\n  <!-- Properties -->";
    for (const item of code.items) {
      newCode += `\n  <property name="${ item.name }" value="\${dockerBaseDir}/bin/${ item.name }" />`;
    }
    newCode += "\n  <!-- Tasks -->";

    for (const item of code.items) {
      newCode += `\n  <target name="${ item.name }"${
        Array.isArray(item.dependencies) && item.dependencies.length > 0 ?
        `\n    depends="${ (item.dependencies || []).join(",") }"` : ""
      }${
        item.description ? `\n    description="${ item.description }"` : ""
      }/>`;
    }
  }
  newCode += "\n</project>";
  return newCode;
};
```

![image](https://user-images.githubusercontent.com/28489774/62125797-6b04e200-b2ce-11e9-99f3-689be88bd01a.png)
![image](https://user-images.githubusercontent.com/28489774/62125939-d3ec5a00-b2ce-11e9-9255-b9517336fade.png)

**Comment by soryy708** [comment]
You're using TypeScript. Perhaps this is related? https://github.com/istanbuljs/nyc/issues/1148

**Comment by coreyfarrell** [comment]
I believe this issue is fixed in nyc 15.0.0.

**Comment by kmdrGroch** [comment]
Unfortunately I don't have access to that repository anymore, so i can't confirm it. I guess I'll close this issue for now.
